tools/windows-node-installer/pkg/cloudprovider/aws/ec2_test.go

```diff
@@ -149,11 +142,47 @@ func TestGetMissingRules(t *testing.T) {
 				getOtherPortRules(WINRM_PORT, myIP),
 				getOtherPortRules(sshPort, myIP),
 				getOtherPortRules(rdpPort, myIP),
-			}},
+			},
+		},
 	}
 
 	for _, tt := range testIO {
 		actual := awsProvider.GetRulesForSgUpdate(myIP, tt.in, vpcCidr)
 		assert.ElementsMatch(t, tt.expected, actual, tt.name, "awsProvider.GetRulesForSgUpdate(), actual values do not match expected")
 	}
 }
+
+// TestExamineRulesInSg tests the ExamineRulesInSg function
+func TestExamineRulesInSg(t *testing.T) {
+	myIP := "144.121.20.163"
+	vpcCidr := "1.1.1.1/32"
+	testIO := []struct {
+		name     string
+		in       []*ec2.IpPermission
+		expected []int64
+	}{
+		// test with nil input of rules
+		{
+			name:     "Nil Input",
+			in:       nil,
+			expected: []int64{WINRM_PORT, rdpPort, sshPort},
+		},
+		// test with partial input of rules
+		{
+			name: "Partial input",
+			in: []*ec2.IpPermission{
+				getOtherPortRules(WINRM_PORT, myIP),
+			},
+			expected: []int64{rdpPort, sshPort},
+		},
+	}
+
+	for _, tt := range testIO {
+		actual, _ := ExamineRulesInSg(myIP, tt.in, vpcCidr)
+		if len(actual) != len(tt.expected) {
+			t.Error("ExamineRulesInSg(),lengths of actual and expected slices do not match")
+			break
+		}
+		assert.ElementsMatch(t, tt.expected, actual, tt.name, "ExamineRulesInSg(),actual values do not match expected")
```

Comment:
Wouldn't ElementsMatch do a length check?

Reply:
@ravisantoshgudimetla It would, I was trying to check if the lengths match and avoid the elements matching if they do not match. But I do realize that if the lengths match, it would anyways have to go through the asset method call. So I can edit this to just make one assert call in both cases.